File: scratch/find_lines.py
import os

fpath = 'control.js'
keywords = ['ctrl-btn-pipeline', 'DOMContentLoaded', 'ctrl-btn-bulk-assets']

try:
    with open(fpath, 'r', encoding='utf-8') as f:
        for i, line in enumerate(f, 1):
            for kw in keywords:
                if kw in line:
                    print(f"{i} [{kw}] -> {line.strip()}")
except Exception as e:
    print("Error:", e)

File: scratch/find_lines.js
import fs from 'fs';
import readline from 'readline';

const fpath = 'control.js';
const keywords = ['ctrl-btn-pipeline', 'DOMContentLoaded', 'ctrl-btn-bulk-assets'];

async function search() {
    const fileStream = fs.createReadStream(fpath);
    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    let i = 1;
    for await (const line of rl) {
        for (const kw of keywords) {
            if (line.includes(kw)) {
                console.log(`${i} [${kw}] -> ${line.trim()}`);
            }
        }
        i++;
    }
}
search();
